src/js/game/map_chunk.js

```diff
@@ -225,8 +225,28 @@ export class MapChunk {
             }
         }
         if (windmillCount > 1) {
-            subShapes[0] = enumSubShape.rect;
-            subShapes[1] = enumSubShape.rect;
+            weights[enumSubShape.windmill] = 0;
+            if (distanceToOriginInChunks < 15) {
+                // Turn left or right subshapes into a random type that is not windmill
+                const subShape = this.internalGenerateRandomSubShape(rng, weights);
+                if (rng.next() < 0.5) {
+                    subShapes[0] = subShape;
+                    subShapes[1] = subShape;
+                } else {
+                    subShapes[2] = subShape;
+                    subShapes[3] = subShape;
+                }
+            } else {
+                // Turn subshape at 2 random positions into a random type that is not windmill
+                let quad1 = rng.nextIntRange(0,3);
```

Comment:
`nextIntRange()` does not include the upper bound as one of the potential values